tests/unit/src/unit/UnitBuilder.hx

```diff
@@ -86,13 +88,55 @@ class UnitBuilder {
 				return false;
 			}
 		}
-		var e = switch [isFloat(e1) || isFloat(e2), e2.expr] {
-			case [_, EField( { expr:EConst(CIdent("Math" | "math")) }, "POSITIVE_INFINITY" | "NEGATIVE_INFINITY")] if (Context.defined("cpp") || Context.defined("php")):
+		function isAbstract(e, checkPrev = true) {
+			try return switch(Context.follow(Context.typeof(e))) {
+				case TAbstract(_, _):
+					true;
+				case x:
+					false;
+			} catch (err:Dynamic) {
+				if (checkPrev && previousCode.length > 0) {
+					var arr = previousCode.concat([e]);
+					var e = macro $b{arr};
+					return isAbstract(e , false);
+				}
+				return false;
+			}
+		}
+
+		var e = switch [isAbstract(e1) && isAbstract(e2), isFloat(e1) || isFloat(e2), e2.expr] {
+			case [_, _, EField( { expr:EConst(CIdent("Math" | "math")) }, "POSITIVE_INFINITY" | "NEGATIVE_INFINITY")] if (Context.defined("cpp") || Context.defined("php")):
 				macro t($e1 == $e2);
-			case [true, _]:
+			case [_,true, _]:
 				macro feq($e1, $e2);
+			case [true, _, _]:
+				var same = try {
+					var t1 = Context.follow(Context.typeof(e1));
+					var t2 = Context.follow(Context.typeof(e2));
+					haxe.macro.TypeTools.toString(t1) == haxe.macro.TypeTools.toString(t2);
+				} catch (e:Dynamic) { false; };
+				if (same) {
+
+					#if cppia
+					// calling eqAbstract for Int leads to a segmentation fault with cppia, but only when compiled
+					// with -D nocppiaast
+					var t1 = Context.follow(Context.typeof(e1));
+					var t1 = haxe.macro.TypeTools.toString(t1);
+					if (t1 == "Int") {
+						macro eq($e1, $e2);
+					} else {
+						//macro eq($e1, $e2);
+						macro eqAbstract($e1 == $e2, $e1, $e2);
+					}
```

Comment:
@hughsando can you take a look at this, i have to call `eq` when the type is Int for cppia, but only with `-D nocppiaast`. if i don't do that it leads to a segfault, but it's only in this special circumstance.

Reply:
The nocppiaast is for "legacy test mode" and I don't really want to fix bugs in this code.
Basically, you should not need to use nocppiaast on new code.
Is is acceptable to use this define to hack out the test, or use a "legacy eq" method?

Reply:
I can leave it as it is now, but it's totally strange why this happens at all. I mean it just compares integers, i also traced the values, nothing special but at some point, after the first ~90 comparisons it just segfaults.